posthog/tasks/exporter.py

```diff
@@ -0,0 +1,122 @@
+    options = Options()
+    options.headless = True
+    options.add_argument("--force-device-scale-factor=2")
+
+    return webdriver.Chrome(
+        service=Service(ChromeDriverManager(chrome_type=ChromeType.CHROMIUM, log_level=logging.ERROR).install()),
+        options=options,
+    )
+
+
+def _export_to_png(exported_asset: ExportedAsset) -> None:
+    """
+    Exporting an Insight means:
+    1. Loading the Insight from the web app in a dedicated rendering mode
+    2. Waiting for the page to have fully loaded before taking a screenshot to disk
+    3. Loading that screenshot into memory and saving the data representation to the relevant Insight
+    4. Cleanup: Remove the old file and close the browser session
+    """
+
+    _start = time.time()
+
+    driver = None
+    image_path = None
+
+    try:
+        if not settings.SITE_URL:
+            raise Exception(
+                "The SITE_URL is not set. The exporter must have HTTP access to the web app in order to work"
+            )
+
+        image_id = str(uuid.uuid4())
+        image_path = os.path.join(TMP_DIR, f"{image_id}.png")
+
+        url_to_render = None
+        screenshot_width = 800
+
+        if not os.path.exists(TMP_DIR):
+            os.makedirs(TMP_DIR)
+
+        if exported_asset.insight is not None:
+            url_to_render = f"{settings.SITE_URL}/exporter/{exported_asset.access_token}"
+            wait_for_css_selector = ".ExportedInsight"
+            screenshot_width = 800
+
+        elif exported_asset.dashboard is not None:
+            url_to_render = f"{settings.SITE_URL}/exporter/{exported_asset.access_token}"
+            wait_for_css_selector = ".InsightCard"
+            screenshot_width = 1920
+        else:
+            raise Exception(f"Export is missing required dashboard or insight ID")
+
+        logger.info(f"Exporting {exported_asset.id} from {url_to_render}")
+
+        driver = get_driver()
+        driver.set_window_size(screenshot_width, screenshot_width * 0.5)
+        driver.get(url_to_render)
+
+        # TODO: Change this to something deterministic like a JS object check from the page that all insights have settled
+        WebDriverWait(driver, 10).until(lambda x: x.find_element(By.CSS_SELECTOR, wait_for_css_selector))
+
+        height = driver.execute_script("return document.body.scrollHeight")
+
+        driver.set_window_size(screenshot_width, height)
+        driver.save_screenshot(image_path)
+
+        with open(image_path, "rb") as image_file:
+            image_data = image_file.read()
+
+        exported_asset.content = image_data
+        exported_asset.save()
+
+        os.remove(image_path)
+
+        logger.info(f"Export took: {time.time() - _start}s")
+
+    except Exception as err:
+        # Ensure we clean up the tmp file in case anything went wrong
+        if image_path and os.path.exists(image_path):
+            os.remove(image_path)
+
+        logger.error(f"Error: {err}")
```

Comment:
might be useful to have a statsd success and error increment so we can graph success vs failure